stonesoup/simulator/transition.py

```diff
@@ -0,0 +1,275 @@
+            transition_times.append(timedelta(seconds=t1))
+            transition_models.append(turn_model)
+
+        dx = x_coord - state.state_vector[0]  # get remaining distance to next x-coord
+        dy = y_coord - state.state_vector[2]  # get remaining distance to next y-coord
+
+        d = np.sqrt(dx**2 + dy**2)  # remaining distance to next coord
+
+        t2 = (time - state.timestamp).total_seconds()  # time remaining before platform should
+        # be at next coord
+
+        if d > 0:  # if platform is not already at target coord, add linear acceleration model
+
+            accel_model = Point2PointConstantAcceleration(state=state,
+                                                          destination=(x_coord, y_coord),
+                                                          duration=timedelta(seconds=t2))
+            state.state_vector = accel_model.function(state=state,
+                                                      time_interval=timedelta(seconds=t2))
+            state.timestamp += timedelta(seconds=t2)
+            transition_times.append(timedelta(seconds=t2))
+            transition_models.append(accel_model)
+
+    return transition_models, transition_times
+
+
+class Point2PointConstantAcceleration:
+    r"""Constant acceleration transition model for 2D cartesian coordinates
+
+    The platform is assumed to move with constant acceleration between two given cartesian
+    coordinates.
+    Motion is determined by the kinematic formulae:
+
+        .. math::
+            v &= u + at \\
+            s &= ut + \frac{1}{2} at^2
+
+    Where :math:`u, v, a, t, s` are initial speed, final speed, acceleration, transition time and
+    distance travelled respectively.
+    If the required constant acceleration leads to the platform overshooting (and coming back to)
+    the destination coordinates, a :class:`~.Point2PointStop` transition model is called instead.
+    """
+
+    def __init__(self, state, destination, duration):
+        dx = destination[0] - state.state_vector[0]  # x-distance to destination
+        dy = destination[1] - state.state_vector[2]  # y-distance to destination
+        ux = state.state_vector[1]  # initial x-speed
+        uy = state.state_vector[3]  # initial y-speed
+
+        t = duration.total_seconds()  # duration of acceleration
+
+        self.ax = 2*(dx - ux*t) / t**2  # x-acceleration
+        self.ay = 2*(dy - uy*t) / t**2  # y-acceleration
+
+        vx = ux + self.ax*t  # final x-speed
+        vy = uy + self.ay*t  # final y-speed
+
+        self.over_shot = False
+
+        if np.sign(ux) != np.sign(vx) or np.sign(uy) != np.sign(vy):
+            # if the resultant speeds have changed sign, the platform must have turned-back on
+            # itself hence must have overshot the destination
+            self.over_shot = True
+            self.new_model = Point2PointStop(state, destination, duration)  # use a different model
```

Comment:
This  `new_model` construction seems a little inelegant. Unfortunately I can't think of a way brilliant way around it... Perhaps a "factory function" that generates the required model and returns either one or the other?

Reply:
Agreed. Okay as a placeholder, but something that could be cleaned at a later date?

Reply:
What do you think of the approach in #298? If you like it, I think you can just merge into this PR, before this one is merged.